apps/android/app/src/main/java/chat/simplex/app/views/usersettings/UserAddressView.kt

```diff
@@ -1,118 +1,398 @@
+      onCloseHandler,
+      createAddress = {
+        withApi {
+          progressIndicator = true
+          val connReqContact = chatModel.controller.apiCreateUserAddress()
+          if (connReqContact != null) {
+            chatModel.userAddress.value = UserContactLinkRec(connReqContact)
+
+            AlertManager.shared.showAlertDialog(
+              title = generalGetString(R.string.delete_address_with_contacts_question),
+              text = generalGetString(R.string.add_address_to_your_profile),
+              confirmText = generalGetString(R.string.share_verb),
+              onConfirm = {
+                setProfileAddress(true)
+                shareViaProfile.value = true
+              }
+            )
           }
-        },
-        destructive = true,
+          progressIndicator = false
+        }
+      },
+      learnMore = {
+        ModalManager.shared.showModal {
+          Column(
+            Modifier
+              .fillMaxHeight()
+              .padding(horizontal = DEFAULT_PADDING),
+            verticalArrangement = Arrangement.SpaceBetween
+          ) {
+            UserAddressLearnMore()
+          }
+        }
+      },
+      share = { userAddress: String -> shareText(cxt, userAddress) },
+      setProfileAddress = ::setProfileAddress,
+      deleteAddress = {
+        AlertManager.shared.showAlertDialog(
+          title = generalGetString(R.string.delete_address__question),
+          text = if (shareViaProfile.value) generalGetString(R.string.all_your_contacts_will_remain_connected_update_sent) else generalGetString(R.string.all_your_contacts_will_remain_connected),
+          confirmText = generalGetString(R.string.delete_verb),
+          onConfirm = {
+            progressIndicator = true
+            withApi {
+              val u = chatModel.controller.apiDeleteUserAddress()
+              if (u != null) {
+                chatModel.userAddress.value = null
+                chatModel.updateUser(u)
+                shareViaProfile.value = false
+                progressIndicator = false
+              }
+            }
+          },
+          destructive = true,
+        )
+      },
+      saveAas = { aas: AutoAcceptState, savedAAS: MutableState<AutoAcceptState> ->
+        withBGApi {
+          val address = chatModel.controller.userAddressAutoAccept(aas.autoAccept)
+          if (address != null) {
+            chatModel.userAddress.value = address
+            savedAAS.value = aas
+          }
+        }
+      })
+  }
+
+  if (viaCreateLinkView) {
+    showLayout()
+  } else {
+    ModalView(close = { onCloseHandler.value(close) }) {
+      showLayout()
+    }
+  }
+
+  if (progressIndicator) {
+    Box(
+      Modifier.fillMaxSize(),
+      contentAlignment = Alignment.Center
+    ) {
+      CircularProgressIndicator(
+        Modifier
+          .padding(horizontal = 2.dp)
+          .size(30.dp),
+        color = MaterialTheme.colors.secondary,
+        strokeWidth = 3.dp
```

Comment:
better to add semitransparent white circle underneath when userAddress != null - currently it's hard to see on top of qr code

Reply:
added circle with background color with alpha 0.9f